api/app/ValueObjects/ProfileSnapshot.php

```diff
@@ -142,7 +172,9 @@ private function parseSnapshotEnum(mixed $value, $enum): mixed
     private function parseSnapshotRecursive(RecursiveArrayIterator $rai, array $accumulator, $enumMap)
     {
         foreach ($rai as $k => $v) {
-            if (array_key_exists($k, $enumMap)) {
+            if (($k === 'classification' || $k === 'department') && isset($v)) {
```

Comment:
I don't love listing individual properties here - it's pretty brittle.  Do we have a programmatic way of identifying these?

Reply:
It was tricky for me to figure out how to get it to work at all 😆 
I've no clue for best practices

Reply:
OK, I took a closer look at this and I'm wondering if this logic needs to exist at all.  Why can't the recursive parser just treat these like other related models?  In the snapshot they don't look any different.  I wonder if perhaps you're working around a bug in the storing?
![image](https://github.com/user-attachments/assets/565daf00-f412-403c-8169-fe728d335f21)
Shouldn't these look like?
```json
"department": {
     "id": "3241d6c5-2cdc-4c94-9ce4-28b610100098"
},
"classification": {
     "id": "1efbd3ea-ad55-4c7b-bfea-2feca5df1aca"
}
```

Reply:
Kind of related to the question about DepartmentBelongsTo

Reply:
Okay, I think I got it! Seems like I needed to use the Resource files instead 
https://github.com/GCTC-NTGC/gc-digital-talent/pull/11947/commits/6f83723b5d8b7be9007a66ac1b368f210862c1a5

Reply:
Nice, this is looks way better!